File: src/main/java/io/github/nikpivkin/LifetimeTaskManager.java
package io.github.nikpivkin;

import java.util.Collection;
import org.bukkit.entity.Player;

public interface LifetimeTaskManager {

  int ONE_SECOND_IN_TICKS = 20;

  void startupTask(Player player);

  void startupTask(Collection<? extends Player> players);

  void cancelTask(Player player);

  void cancelTask(Collection<? extends Player> players);
}

File: src/main/kotlin/io/github/nikpivkin/lifetime/LifetimeTaskManager.kt
package io.github.nikpivkin.lifetime

import org.bukkit.entity.Player

interface LifetimeTaskManager {
    fun startupTask(player: Player)
    fun startupTask(players: Collection<Player>)
    fun cancelTask(player: Player)
    fun cancelTask(players: Collection<Player>)

    companion object {
        const val ONE_SECOND_IN_TICKS = 20L
    }
}
